src/Microsoft.AspNetCore.Mvc.Razor.Host/MvcRazorParser.cs

```diff
@@ -87,29 +87,7 @@ public class MvcRazorParser : RazorParser
                 _viewImportsDirectiveDescriptors,
                 errorSink);
 
-            var descriptors = visitor.GetDescriptors(documentRoot);
-            foreach (var descriptor in descriptors)
-            {
-                foreach (var attributeDescriptor in descriptor.Attributes)
-                {
-                    if (attributeDescriptor.IsIndexer &&
-                        string.Equals(
-                            attributeDescriptor.TypeName,
-                            _modelExpressionTypeName,
-                            StringComparison.Ordinal))
-                    {
-                        errorSink.OnError(
-                            SourceLocation.Undefined,
-                            Resources.FormatMvcRazorParser_InvalidPropertyType(
-                                descriptor.TypeName,
-                                attributeDescriptor.Name,
-                                _modelExpressionTypeName),
```

Comment:
Remove the `_modelExpressionTypeName` field from this class. Leave the constructor parameter and its `null` check alone but don't use it.